php56.extensions.tidy broken

```ShellSession
$ nix build .#php56.extensions.tidy -L
[…]
php-tidy> building
php-tidy> build flags: -j8 -l8 SHELL=/nix/store/pbfraw351mksnkp2ni9c4rkc9cpp89iv-bash-5.1-p12/bin/bash
php-tidy> /nix/store/pbfraw351mksnkp2ni9c4rkc9cpp89iv-bash-5.1-p12/bin/bash /build/php-5.6.40/ext/tidy/libtool --mode=compile gcc  -I. -I/build/php-5.6.40/ext/tidy -DPHP_ATOM_INC -I/build/php-5.6.40/ext/tidy/include -I/build/php-5.6.40/ext/tidy/main -I/build/php-5.6.40/ext/tidy -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/main -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/TSRM -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/Zend -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext/date/lib -I/nix/store/67z2xvjmwial7fmii6cr8qpf4l334zyl-html-tidy-5.8.0/include  -DHAVE_CONFIG_H  -g -O2   -c /build/php-5.6.40/ext/tidy/tidy.c -o tidy.lo
php-tidy> mkdir .libs
php-tidy>  gcc -I. -I/build/php-5.6.40/ext/tidy -DPHP_ATOM_INC -I/build/php-5.6.40/ext/tidy/include -I/build/php-5.6.40/ext/tidy/main -I/build/php-5.6.40/ext/tidy -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/main -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/TSRM -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/Zend -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext/date/lib -I/nix/store/67z2xvjmwial7fmii6cr8qpf4l334zyl-html-tidy-5.8.0/include -DHAVE_CONFIG_H -g -O2 -c /build/php-5.6.40/ext/tidy/tidy.c  -fPIC -DPIC -o .libs/tidy.o
php-tidy> /build/php-5.6.40/ext/tidy/tidy.c:34:10: fatal error: buffio.h: No such file or directory
php-tidy>    34 | #include "buffio.h"
php-tidy>       |          ^~~~~~~~~~
php-tidy> compilation terminated.
php-tidy> make: *** [Makefile:181: tidy.lo] Error 1
```

Noticed in https://github.com/fossar/selfoss/runs/4924028211?check_suite_focus=true

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -64,6 +64,9 @@ jobs:
       - name: Build Xdebug extension
         run: nix-build -A outputs.packages.x86_64-linux.php${{ matrix.php.major }}${{ matrix.php.minor }}.extensions.xdebug
 
+      - name: Build Tidy extension
+        run: nix-build -A outputs.packages.x86_64-linux.php${{ matrix.php.major }}${{ matrix.php.minor }}.extensions.tidy
+
       - name: Check that composer PHAR works
         run: |
           nix-shell -E '

--- a/pkgs/package-overrides.nix
+++ b/pkgs/package-overrides.nix
@@ -287,6 +287,20 @@ in
       else
         throw "php.extensions.redis3 requires PHP version < 8.0.";
 
+    tidy = prev.extensions.tidy.overrideAttrs (attrs: {
+      patches =
+        let
+          upstreamPatches =
+            attrs.patches or [];
+
+          ourPatches =
+            lib.optionals (lib.versionOlder prev.php.version "7.1") [
+              ./tidy-html5.patch
+            ];
+        in
+        ourPatches ++ upstreamPatches;
+    });
+
     xdebug =
       # xdebug versions were determined using https://xdebug.org/docs/compat
       if lib.versionAtLeast prev.php.version "7.2" then

--- a/pkgs/tidy-html5.patch
+++ b/pkgs/tidy-html5.patch
@@ -1,0 +1,21 @@
+From: =?utf-8?q?Ond=C5=99ej_Sur=C3=BD?= <ondrej@sury.org>
+Date: Fri, 24 Jun 2016 08:21:28 +0200
+Subject: Adjust tidy extension for tidy-html5
+
+---
+ ext/tidy/tidy.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/ext/tidy/tidy.c b/ext/tidy/tidy.c
+index 4e307eec0..8cecf1795 100644
+--- a/ext/tidy/tidy.c
++++ b/ext/tidy/tidy.c
+@@ -31,7 +31,7 @@
+ #include "ext/standard/info.h"
+
+ #include "tidy.h"
+-#include "buffio.h"
++#include "tidybuffio.h"
+
+ /* compatibility with older versions of libtidy */
+ #ifndef TIDY_CALL

